openmdao/core/problem.py

```diff
@@ -138,8 +144,30 @@ def __init__(self, model=None, driver=None, comm=None, name=None, **options):
         """
         Initialize attributes.
         """
+        global _problem_counter
+
         self.cite = CITATION
-        self._name = name
+
+        # Code to give non-empty names to Problems so that they can be
+        # referenced from command line tools (e.g. check) that accept a Problem argument
+        _problem_counter += 1
```

Comment:
Do we actually need `_problem_counter`, if the number of problems is always the length of `_problem_names`?

Reply:
Also a great point. Will modify code to do that